proposals/0387-cross-compilation-destinations.md

```diff
@@ -167,75 +180,174 @@ that, a separate `toolset.json` file is introduced:
+supports](https://github.com/apple/swift-package-manager/pull/1098) and `destination.json` files presented in the pitch
+version of this proposal, hence `"schemaVersion": "3.0"`. We'll keep parsing `"version": 1` and `"version": 2` for
+backward compatibility, but for consistency with `info.json` this field is renamed to `"schemaVersion"`. Here's an
+informally defined schema for these files:
+
+```json5
+{
+  "schemaVersion": "3.0",
+  "runTimeTriples": [
+    "<triple1>": {
+      "sdkRootPath": "<a required path relative to `destination.json` containing SDK root>",
+      // all of the properties listed below are optional:
+      "swiftResourcesPath": "<a path relative to `destination.json` containing Swift resources for dynamic linking>",
+      "swiftStaticResourcesPath": "<a path relative to `destination.json` containing Swift resources for static linking>",
+      "includeSearchPaths": ["<array of paths relative to `destination.json` containing headers>"],
+      "librarySearchPaths": ["<array of paths relative to `destination.json` containing libraries>"],
+      "toolsetPaths": ["<array of paths relative to `destination.json` containing toolset files>"]
+    },
+    // a destination can support more than one run-time triple:
+    "<triple2>": {
+      "sdkRootPath": "<a required path relative to `destination.json` containing SDK root>",
+      // all of the properties listed below are optional:
+      "swiftResourcesPath": "<a path relative to `destination.json` containing Swift resources for dynamic linking>",
+      "swiftStaticResourcesPath": "<a path relative to `destination.json` containing Swift resources for static linking>",
+      "includeSearchPaths": ["<array of paths relative to `destination.json` containing headers>"],
+      "librarySearchPaths": ["<array of paths relative to `destination.json` containing libraries>"],
+      "toolsetPaths": ["<array of paths relative to `destination.json` containing toolset files>"]
+    }
+    // more triples can be supported by a single destination if needed, primarily for sharing files between them.
+  ]
+}
+```
+
+We propose that all relative paths in `destination.json` files should be validated not to "escape" the destination
+bundle for security reasons, in the same way that `toolset.json` files are validated when contained in destination
+bundles. That is, `../` components, if present in paths, will not be allowed to reference files and
+directories outside of a corresponding destination bundle. Symlinks will also be validated to prevent them from escaping
+out of the bundle.
+ 
+If `sdkRootPath` is specified and `swiftResourcesPath` is not, the latter is inferred to be
+`"\(sdkRootPath)/usr/lib/swift"` when linking the Swift standard library dynamically, `"swiftStaticResourcesPath"` is
+inferred to be `"\(sdkRootPath)/usr/lib/swift_static"` when linking it dynamically. Similarly, `includeSearchPaths` is
```

Comment:
The second time: dynamically -> statically. Also, won't these be used just to build static libraries too, so maybe it should be made more general to `when linking libraries dynamically` in the first instance?

Reply:
Thanks, that's addressed now.

> won't these be used just to build static libraries too

I'm not aware of that. Can you point to specific place in the source code or documentation that supports this statement?

Reply:
I just built SwiftSyntax, which has several static libraries, on linux to check, and it appears to build the object files in a SPM target by looking in the dynamic resource directory for the stdlib interface files, then simply use `ar` to package them together. I don't know if that's a mistake, as the static resource directory has its own interface files, but what you wrote about the static stdlib initially is correct.